examples/web_aspect_ratio.rs

```diff
@@ -0,0 +1,107 @@
+pub fn main() {
+    println!("This example must be run with cargo run-wasm --example web_aspect_ratio")
+}
+
+#[cfg(target_arch = "wasm32")]
+mod wasm {
+    use wasm_bindgen::prelude::*;
+    use wasm_bindgen::JsCast;
+    use web_sys::{HtmlCanvasElement, HtmlElement};
+    use winit::{
+        dpi::PhysicalSize,
+        event::{Event, WindowEvent},
+        event_loop::{ControlFlow, EventLoop},
+        window::{Window, WindowBuilder},
+    };
+
+    const EXPLANATION: &str = "
+This example draws a circle in the middle of a 4/1 aspect ratio canvas which acts as a useful demonstration of winit's resize handling on web.
+Even when the browser window is resized or aspect-ratio of the div changed the circle should always:
+* Fill the entire width or height of the canvas (whichever is smaller) without exceeding it.
+* Be perfectly round
+* Not be blurry or pixelated (there is no antialiasing so you may still see jagged edges depending on the DPI of your monitor)
+";
+
+    #[wasm_bindgen(start)]
+    pub fn run() {
+        console_log::init_with_level(log::Level::Debug).expect("error initializing logger");
+        let event_loop = EventLoop::new();
+
+        let window = WindowBuilder::new()
+            .with_title("A fantastic window!")
+            // A small default size is used to better demonstrate issues that come from failing to update the size
+            .with_inner_size(PhysicalSize::new(100, 100))
```

Comment:
This could be a bit misleading to people reading the example, since you almost never want to use `with_inner_size` when sizing using `ResizeObserver`; it works in this case since the canvas' `style` gets overridden below, but that should probably be documented more clearly.

Something like: 'Note: this would normally fix the canvas' width and height to 100px, but since the canvas' `style` gets overridden below in `create_canvas`, it just sets the initial framebuffer size.'